src/systems/velocity_control/VelocityControl.cc

```diff
@@ -88,19 +117,47 @@ void VelocityControl::Configure(const Entity &_entity,
     return;
   }
 
-  // Subscribe to commands
-  std::vector<std::string> topics;
+  // Subscribe to model commands
+  std::vector<std::string> modelTopics;
   if (_sdf->HasElement("topic"))
   {
-    topics.push_back(_sdf->Get<std::string>("topic"));
+    modelTopics.push_back(_sdf->Get<std::string>("topic"));
   }
-  topics.push_back("/model/" + this->dataPtr->model.Name(_ecm) + "/cmd_vel");
-  auto topic = validTopic(topics);
+  modelTopics.push_back(
+    "/model/" + this->dataPtr->model.Name(_ecm) + "/cmd_vel");
+  auto modelTopic = validTopic(modelTopics);
   this->dataPtr->node.Subscribe(
-    topic, &VelocityControlPrivate::OnCmdVel, this->dataPtr.get());
-
-  ignmsg << "VelocityControl subscribing to twist messages on [" << topic << "]"
+    modelTopic, &VelocityControlPrivate::OnCmdVel, this->dataPtr.get());
+  ignmsg << "VelocityControl subscribing to twist messages on ["
+         << modelTopic << "]"
          << std::endl;
```

Comment:
This seems to be slightly different from the `diff` I am seeing here: https://github.com/ignitionrobotics/ign-gazebo/compare/ign-gazebo4...ign-gazebo3

It looks like multiple topics are now being supported instead of just one. Is this because of new features in `ign-gazebo4` that are to be supported, while being backwards compatible with `ign-gazebo3` functionality?

Reply:
I believe the valid topic was just implemented differently on each branch, getting the similar results.

On #427, targeted at `ign-gazebo3`, this pattern was used:

```
modelTopic = defaultTopic
if (sdfTopic)
  modelTopic = sdfTopic
modelTopic = valid(modelTopic)
```

On #522, targeted at `ign-gazebo4`, the following pattern was used. This is slightly better because it will fallback to a valid default topic if the SDF topic can't be made valid (warning as needed).

```
topics = {sdfTopic, defaultTopic}
topic = firstValidTopic(topics)
```

So as we merge forward, I kept the approach that was already on `ign-gazebo4`, while changing the topic name to `modelTopic` to differentiate from the new `linkTopic`.

Reply:
That function itself can be backported, but it's of little value on `ign-gazebo3` because the `Name` component only supports spaces from Dome, see https://github.com/ignitionrobotics/ign-gazebo/pull/244